### ISSUE TITLE
cloud_types: SMBC, TOBJ and G4BA are reported although SAP's own cloud generators produce them

### ISSUE THREAD

**Issue by jmoutreux-twm** [abaplint/abaplint#4229]
## Symptom

With `"version": { "release": "v793", "language": "Cloud" }`, a package generated
by SAP's ADT *Business Configuration Maintenance Object* wizard reports:

```
src\zemailapplication.smbc.json[1, 1]      - Object type SMBC not supported in cloud (cloud_types) [E]
src\zemailapplicationt.tobj.xml[1, 1]      - Object type TOBJ not supported in cloud (cloud_types) [E]
src\zui_emailapplication_o4.g4ba.xml[1, 1] - Object type G4BA not supported in cloud (cloud_types) [E]
abaplint: 3 issue(s) found, 66 file(s) analyzed
```

The package is the unmodified output of the wizard, linted with
`{"rules": {"cloud_types": true}}` alone.

## Why all three are legitimate in a cloud package

None of these objects was hand-written. All three are produced by SAP tooling
**into an ABAP for Cloud package**, and all activate there:

* **SMBC** — the business configuration maintenance object itself, the very
  artefact the ADT *Business Configuration Maintenance Object* generator exists to
  produce. It holds the service configuration and the whole screen layout of the
  generated Fiori application. Reporting it as unsupported in cloud is the most
  surprising of the three, since it *is* the cloud-native way to deliver
  configuration.
* **TOBJ** — the same generator asks for a transport object name and creates one
  of type *Individual Transaction*, under which configuration changes are
  recorded. It is a mandatory part of the generated business configuration object.
* **G4BA** — created together with the OData V4 service binding.

Observed on S/4HANA 2023 (SAP_BASIS 758) and S/4HANA 2025, in packages whose ABAP
language version is *ABAP for Cloud Development*.

## Root cause

`CloudTypes.isCloud()` in `packages/core/src/rules/cloud_types.ts` is an explicit
allow-list of object classes. It contains `BusinessCatalog`,
`BusinessCatalogAppAssignment`, `ServiceBinding` and `ServiceDefinition`, but not
the three classes above — even though abaplint already models all three:

* `packages/core/src/objects/business_configuration_maintenance_object.ts`
  → `export class BusinessConfigurationMaintenanceObject`, `getType()` → `"SMBC"`
* `packages/core/src/objects/maintenance_and_transport_object.ts`
  → `export class MaintenanceAndTransportObject`
* `packages/core/src/objects/gateway_odata_group_and_assignment.ts`
  → `export class GatewayODataGroupAndAssignment`

So the object types are known to abaplint; they are simply missing from the list.
Note the alphabetical gap in `isCloud()`: it jumps straight from
`BusinessCatalogAppAssignment` to `CDSMetadataExtension`.

This is the same situation as #2998, where CHKC/CHKO/CHKV were reported although
they had been introduced precisely to handle ATC checks in a cloud context.

## Proposed fix

Three entries in `CloudTypes.isCloud()`:

```ts
      || obj instanceof Objects.BusinessConfigurationMaintenanceObject
      || obj instanceof Objects.MaintenanceAndTransportObject
      || obj instanceof Objects.GatewayODataGroupAndAssignment
```

## Environment

* abaplint 2.120.29 (`@abaplint/cli@latest`)
* config: `{ "release": "v793", "language": "Cloud" }`
* objects generated by the ADT Business Configuration Maintenance Object wizard

**Comment by larshp** [comment]
try again, latest version
